types/status.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2019 The MayaData Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConditionType is a custom datatype that refers to
+// various conditions supported by this operator.
+type ConditionType string
```

Comment:
IMO let us avoid getting too deep into conditions. I feel we don't need them unless we create dependencies amongst controllers. We will rely more on events for debugging.